src/api/ledgerStorage/github.js

```diff
@@ -67,7 +89,75 @@ export class GithubStorage implements LedgerStorage {
     return Ledger.parse(rawLedger);
   }
 
-  async write() {
-    throw new Error("method not implemented");
+  async write(ledger: Ledger) {
+    const ledgerData = ledger.serialize();
+
+    // Get latest commit hash of target branch
+    const refRes = await fetch(
+      `${this._getRepoEndpoint()}/git/ref/heads/${this._branch}`,
+      {
+        method: "GET",
+        headers: {
+          Authorization: `Bearer ${this._token}`,
+        },
+      }
+    );
+
+    const refResJson = await refRes.json();
+    const baseCommit = refResJson.object.sha;
+
+    // Create a new tree from the latest commit and update the ledger blob
+    const ledgerBlobRes = await fetch(`${this._getRepoEndpoint()}/git/trees`, {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${this._token}`,
+      },
+      body: JSON.stringify({
+        tree: [
+          {
+            content: ledgerData,
+            type: "blob",
+            path: "data/ledger.json",
+            mode: "100644", // see https://docs.github.com/en/rest/reference/git#create-a-tree
+          },
+        ],
+        base_tree: baseCommit,
+      }),
+    });
+
+    const blobResJson: CreateBlobRes = await ledgerBlobRes.json();
+
+    // Create a commit with the new tree on top of the target branch
+    const commitRes = await fetch(`${this._getRepoEndpoint()}/git/commits`, {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${this._token}`,
+      },
+      body: JSON.stringify({
+        message: "Ledger Update",
```

Comment:
Could be nice to accept an optional string that could make the commit message more descriptive.